### PR TITLE
Optimize dictionary blocks and orc tests

### PR DESCRIPTION
**Pull request by cberner**
This reduces the time for the full orc tests by ~20%

### CODE REVIEW

**Comment by cberner** [comment]
Looks like there's a test failure

File: presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java [hunk 1]
```diff
@@ -85,14 +85,12 @@ public DictionaryBlock(int positionCount, Block dictionary, Slice ids, boolean d
...
-                    if (!dictionary.isNull(position)) {
-                        sizeInBytes += dictionary.getLength(position);
-                    }
-                    uniqueIds++;
+            for (int i = 0; i < positionCount; i++) {
+                int position = getIndex(ids, i);
+                if (!dictionary.isNull(position)) {
+                    sizeInBytes += dictionary.getLength(position);
                 }
+                uniqueIds++;
```
Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java]: This change doesn't seem correct or at least changes the definition of sizeInBytes and uniqueIds.  The new code is counting the all positions even if they refer to the same dictionary entry, so anything that is relying on these numbers will need to be adjusted.

@nileema should review this

Comment (cberner) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java]: Yep, I realized that's why the tests are failing. I'll update it and assign to Nileema

**Comment by nileema** [comment]
Looks good.

**Pull request merged**